pymc3/step_methods/hmc/base_hmc.py

```diff
@@ -41,7 +43,8 @@ def __init__(self, vars=None, scaling=None, step_scale=0.25, is_cov=False,
         vars = inputvars(vars)
 
         if scaling is None and potential is None:
-            scaling = model.test_point
+            bij = DictToArrayBijection(ArrayOrdering(vars), model.test_point)
+            scaling = np.ones(bij.map(model.test_point).size)
```

Comment:
It might be faster to do:
```python
scaling = np.ones(model.dict_to_array(model.test_point).size)
```
which don't need to import DictToArrayBijection, ArrayOrdering as well